theme/lumo/vaadin-horizontal-layout-styles.html

```diff
@@ -52,6 +52,61 @@
         content: "";
         margin-left: calc(var(--lumo-space-xl) * -1);
       }
+
+      /* RTL styles */
+      :host([dir="rtl"][theme~="spacing-xs"]) ::slotted(*) {
+        margin-left: inherit;
```

Comment:
Copy-pasting this over and over for evert theme does not seem right to me.

What I would do instead:

1. Group all the `::before` declarations:

```css
:host([theme~="spacing-xs"])::before,
:host([theme~="spacing-s"])::before, 
:host([theme~="spacing"])::before, 
:host([theme~="spacing-l"])::before, 
:host([theme~="spacing-l"])::before, 
:host([theme~="spacing-xl"])::before {
  content: "";
} 
```

2. Update definitions to use `:not` like that:

```css
:host([theme~="spacing-xs"]:not([dir="rtl"]))::before {
  margin-left: var(--lumo-space-xs);
}

:host([theme~="spacing-xs"][dir="rtl"])::before {
  margin-right: var(--lumo-space-xs);
}
```

3. Same thing for `::slotted(*)` blocks of rules.

This way we avoid using `inherit`.

Note, the RTL rules are still better to group separately at the bottom of the files.

Reply:
Yeah. It makes total sense and it's much better this way. Changed.